graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/federation/DefaultDgsFederationResolver.kt

```diff
@@ -87,15 +93,15 @@ open class DefaultDgsFederationResolver() :
                     val fetcher = entityFetcherRegistry.entityFetchers[typename]
                         ?: throw MissingDgsEntityFetcherException(typename.toString())
 
-                    if (!fetcher.second.parameterTypes.any { it.isAssignableFrom(Map::class.java) }) {
```

Comment:
Yeah, that's a good point. I think it might need to be adjusted a bit to improve visibility into type expectations, though, otherwise it could hide incorrectly implemented `@DgsEntityRepresentationMapper` implementations. 

I was thinking this:

1. If we _do_ have a representation mapper for this representation...
  a) check if the mapper accepts a parameter of type `Map::class.java`, and throw an `InvalidDgsEntityRepresentationMapper` if it does not.
  b) check if the entity fetcher provides a parameter that is compatible with the mapped representation type, and throw an `InvalidDgsEntityFetcher` indicating that it should accept the mapped type if it does not.
2. If we _do not_ have a representation mapper for this representation...
  a) check if the entity fetcher provides a parameter that is compatible with `Map::class.java`, and throw an `InvalidDgsEntityRepresentationMapper` indicating that it should accept a `Map::class.java` if it does not (current behavior).

Since we can assume that no one will be using the new `@DgsEntityRepresentationMapper` annotation, this should still be backwards compatible with the current version, if I'm understanding correctly. @berngp thoughts?

Reply:
> b) check if the entity fetcher provides a parameter that is compatible with the mapped representation type, and throw an `InvalidDgsEntityFetcher` indicating that it should accept the mapped type if it does not.

Not sure that is the best approach, suggest the check is as follows...
1. Check if the entity fetcher supports a `Map` as representation.
1.1 Proceed with _just passing_ the representation as a map. i.e. as is.
2. If not, fetch the type of the parameter expected by the _entity fetcher_ as _representation_.
3. Find the _Representation Mapper_ that applies for the `__typename` as well as the _type_ expected by the _entity fetcher's parameter_. i.e. the _Representation Mapper_ should have a _method_ that tells us if they _apply_ or not for the `__typename` and the expected class.
3.1 If there are candidate _mappers_, proceed and map the type and call the _entity fetcher_. Log which mapper was used.
4. If there is no _Representation Mapper_ throw an `InvalidDgsEntityRepresentationMapper` that specifies the `__typename` the expected _entity fetcher_ parameter type, and the class and method of the _entity fetcher_.

This will allow us to roll the feature out without affecting any customers. Even if the _Representation Mapper_ is provided, it will only apply for the _Entity Fetchers_ that are ready and explicitly expect the _representation_ as a type other than a `Map`.

Reply:
I like that approach and will take a stab at applying it. Thanks for the feedback so far! 😄

Reply:
@wbaldoumas - Thanks so much for all the work so far. A few of us were able to take a look and discuss the approach. 
We decided to not go ahead with this PR  because 1) Fixing the issue of scalar parsing  would be better handled within the framework ideally. We don't quite yet have a solution but will need to explore it further. 2) We agree it would be nice to have a better way of making the representation available readily for the user within the fetcher, we would like to consider a mechanism that leverages existing code conversions, similar to `@InputArgument` if possible. 

We also agreed that the solution proposed in this PR would be better handled within a library outside of the framework. We really appreciate all the effort and for kicking off the ideation process. Ultimately this is a current gap in the framework, so we will  prioritize adding support for this.

Reply:
We will close out this PR and post an update when we have the new feature.

Reply:
@srinivasankavitha that works for me. Thanks so much for taking a look and for the collaboration so far.

No worries on dropping this PR. I'm glad that there are more eyes on this problem at this point and that some solution for it will be getting prioritized. It has also been great to poke around at the internals here and get an understanding of how some of this works under the hood. 😄 

I'll keep an eye out on the [issue](https://github.com/Netflix/dgs-framework/issues/1569) @gwardwell posted to track progress.

Reply:
Thank you!